crates/trie/sparse/src/trie.rs

```diff
@@ -174,19 +275,25 @@ impl<P: BlindedProvider> SparseTrie<P> {
 /// - All keys in `values` collection are full leaf paths.
 #[derive(Clone, PartialEq, Eq)]
 pub struct RevealedSparseTrie<P = DefaultBlindedProvider> {
-    /// Blinded node provider.
+    /// Provider used for retrieving blinded nodes.
+    /// This allows lazily loading parts of the trie from an external source.
     provider: P,
-    /// All trie nodes.
+    /// Map from a prefix (nibbles) to its corresponding sparse trie node.
+    /// This contains all of the reveal nodes in trie.
     nodes: HashMap<Nibbles, SparseNode>,
-    /// All branch node tree masks.
+    /// Mask information for branch nodes that require tree storage.
+    /// When a branch is set, the corresponding child subtree is stored in the database.
     branch_node_tree_masks: HashMap<Nibbles, TrieMask>,
-    /// All branch node hash masks.
+    /// Mask information for branch nodes that require hash storage.
+    /// When a bit is set, the cooresponding child is stored as a hash in the database.
```

Comment:
a bit confusing what does "tree/hash storage" mean here. Second lines are correct, tree mask is about whether the subtree is stored in the database, and hash mask is about whether the hash for a child is stored in the database.